libtsuba/src/RDGPartHeader.h

```diff
@@ -39,12 +39,24 @@ class KATANA_EXPORT RDGPartHeader {
   // Property manipulation
   //
 
-  void AppendNodePropStorageInfo(PropStorageInfo&& pmd) {
-    node_prop_info_list_.emplace_back(std::move(pmd));
+  void AddNodePropStorageInfo(PropStorageInfo&& pmd) {
+    auto pmd_it = std::find_if(
+        node_prop_info_list_.cbegin(), node_prop_info_list_.cend(),
+        [&](const PropStorageInfo& my_pmd) { return my_pmd.name == pmd.name; });
+    if (pmd_it == node_prop_info_list_.end()) {
+      node_prop_info_list_.emplace_back(std::move(pmd));
+    }
+    // If we already have a record, no need to change it
   }
```

Comment:
@tylershunt, is this true?  It seems like we were appending a record with a valid name, an empty `path` and an uninitialized (and therefore false) `persist `flag.  If `PropStorageInfo `already exists with the right name, I figured I should not change the `path `and `persist` flag.

Reply:
I think it's right not to change the persist flag, but I think the `path` should be cleared somewhere. Currently our semantics are to write properties out when `persist` is true and `path` is empty https://github.com/KatanaGraph/katana/blob/4f418f0aeab539c05fd296f3b28c5b7616dc747f/libtsuba/src/RDG.cpp#L87

The logic being that if  a path exists we can just re-use that since our files are immutable. Here since we are modifying a properties contents I think we should write out the new contents.

Reply:
Thank you!  I missed that.  Will fix.